challenge-manager/chain-watcher/watcher.go

```diff
@@ -395,47 +392,86 @@ func (w *Watcher) ComputeAncestors(
 			challengedAssertionHash,
 		)
 	}
-	blockHeader, err := w.chain.Backend().HeaderByNumber(ctx, w.chain.GetDesiredRpcHeadBlockNumber())
+	blockHeaderNumber, err := w.chain.Backend().HeaderNumberUint64(ctx, w.chain.GetDesiredRpcHeadBlockNumber())
 	if err != nil {
 		return nil, err
 	}
-	if !blockHeader.Number.IsUint64() {
-		return nil, errors.New("block number is not uint64")
+	return chal.honestEdgeTree.ComputeAncestors(ctx, edgeId, blockHeaderNumber)
+}
+
+func (w *Watcher) ClosestEssentialAncestor(
+	ctx context.Context,
+	challengedAssertionHash protocol.AssertionHash,
+	edge protocol.VerifiedRoyalEdge,
+) (protocol.ReadOnlyEdge, error) {
+	chal, ok := w.challenges.TryGet(challengedAssertionHash)
+	if !ok {
+		return nil, fmt.Errorf(
+			"could not get challenge for top level assertion %#x",
+			challengedAssertionHash,
+		)
 	}
-	return chal.honestEdgeTree.ComputeAncestors(ctx, edgeId, blockHeader.Number.Uint64())
+	return chal.honestEdgeTree.ClosestEssentialAncestor(ctx, edge)
 }
 
-func (w *Watcher) PathWeightToClosestEssentialAncestor(
+func (w *Watcher) IsEssentialAncestorConfirmable(
 	ctx context.Context,
+	edge protocol.SpecEdge,
 	challengedAssertionHash protocol.AssertionHash,
-	edge protocol.ReadOnlyEdge,
-) (uint64, error) {
+	confirmationThreshold uint64,
+) (bool, error) {
 	chal, ok := w.challenges.TryGet(challengedAssertionHash)
 	if !ok {
-		return 0, fmt.Errorf(
+		return false, fmt.Errorf(
 			"could not get challenge for top level assertion %#x",
 			challengedAssertionHash,
 		)
 	}
-	blockHeader, err := w.chain.Backend().HeaderByNumber(ctx, w.chain.GetDesiredRpcHeadBlockNumber())
+	blockHeaderNumber, err := w.chain.Backend().HeaderNumberUint64(ctx, w.chain.GetDesiredRpcHeadBlockNumber())
 	if err != nil {
-		return 0, err
-	}
-	if !blockHeader.Number.IsUint64() {
-		return 0, errors.New("block number is not uint64")
+		return false, err
 	}
 	if !chal.honestEdgeTree.HasRoyalEdge(edge.Id()) {
-		return 0, fmt.Errorf("edge with id %#x is not yet tracked locally", edge.Id().Hash)
+		return false, fmt.Errorf("edge with id %#x is not yet tracked locally", edge.Id().Hash)
 	}
 	essentialAncestor, err := chal.honestEdgeTree.ClosestEssentialAncestor(ctx, edge)
 	if err != nil {
-		return 0, err
+		return false, err
 	}
-	return chal.honestEdgeTree.ComputePathWeight(ctx, challengetree.ComputePathWeightArgs{
+	pathWeight, err := chal.honestEdgeTree.ComputePathWeight(ctx, challengetree.ComputePathWeightArgs{
 		Child:    edge.Id(),
 		Ancestor: essentialAncestor.Id(),
-		BlockNum: blockHeader.Number.Uint64(),
+		BlockNum: blockHeaderNumber,
 	})
+	if err != nil {
+		return false, err
+	}
+	return pathWeight >= confirmationThreshold, nil
+}
+
+func (w *Watcher) IsConfirmableEssentialNode(
```

Comment:
I don't like introducing the term "Node" here when we already use that to mean the whole nitro stack, and because we have the word "Edge" already. What would be wrong with calling this `IsConfirmableEssentialEdge`

Reply:
I was just trying to match the definitions under https://www.notion.so/arbitrum/Reimbursing-Mini-Stakes-In-Band-in-BoLD-f7d27046d6304c2ab6933a152b83057d but yes agreed "node" is an isolated term and we should call it edge